### PR TITLE
Improve meson build

### PR DESCRIPTION
**Pull request by gasinvein**
- Split top-level `meson.build` into subdirs
- Make vulkan layer manifest architecture-dependent and install it under different names based on target arch
- Write absolute library path to the manifest so that it could be loaded from wherever it's installed
- Slightly rewrite the Makefile so that it behaves closer to the original, drop other makefiles

### CODE REVIEW

File: config/meson.build [hunk 1]
```diff
@@ -0,0 +1,13 @@
+conf_data = configuration_data()
+conf_data.set('LIBPATH', join_paths(get_option('prefix'),
+                                    get_option('libdir'),
+                                    'lib@0@.so'.format(meson.project_name().to_lower())))
+
+configure_file(input: '@0@.json.in'.format(meson.project_name()),
+               output: '@0@.@1@.json'.format(meson.project_name(), target_machine.cpu_family()),
```
Comment (mcoffin) [inline config/meson.build:7]: This is super smart. I like it!

File: config/vkBasalt.json.in [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
   "file_format_version" : "1.0.0",
   "layer" : {
-    "name": "VK_LAYER_VKBASALT_PostProcess32",
+    "name": "VK_LAYER_VKBASALT_PostProcess",
```
Comment (mcoffin) [inline config/vkBasalt.json.in:4]: This change actually breaks things when the `x86` and `x86_64` targets are installed alongside each other. the layer names must be unique.

**Review by mcoffin — CHANGES_REQUESTED** [review]
See my comment on `config/vkBasalt32.json` above.

The layer names have to stay unique, or it won't load the layer when they're installed side by side.

File: config/vkBasalt.json.in (hunk 2 above)
Comment (gasinvein) [reply config/vkBasalt.json.in:4]: Yeah, indeed, missed it. But not much problem, we can configure it in a similar way as the filename.

**Pull request merged**